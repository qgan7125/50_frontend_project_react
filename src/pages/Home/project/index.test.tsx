import { render, screen} from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import Project from './index';

const props =         {
    "name": "Expanding Cards",
    "backgroundImage": "01ExpandingCards.png",
    "day": 1,
    "link": "01ExpandingCards"
}

describe("Project box test cases", () => {
    test("snapshot test for Project box", async () => {
        const view = (
            render(
                <Router>
                    <Project {...props} />
                </Router>,
            ));
        expect(view.asFragment()).toMatchSnapshot();
    });

    test("All component should be rendered", async () => {
        render(
            <Router>
                <Project {...props} />
            </Router>,
        );

        // Test the day
        const day = screen.queryByText("Day 1");
        expect(day).toBeInTheDocument();
        // Test the title
        const name = screen.queryByText("Expanding Cards");
        expect(name).toBeInTheDocument();
        // Test button
        const buttonEl = screen.queryByText("Live Demo");
        expect(buttonEl).toBeInTheDocument();
        // Test image
        const bgImg = await screen.queryByTestId("project_container");
        expect(bgImg).toHaveStyle(`background-image: url(${process.env.PUBLIC_URL + '/assests/' + props.backgroundImage})`)
    })
})