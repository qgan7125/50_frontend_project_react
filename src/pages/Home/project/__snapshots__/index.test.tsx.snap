// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project box test cases snapshot test for Project box 1`] = `
<DocumentFragment>
  <div
    class="project__container"
    data-testid="project_container"
    style="background-image: url(/assests/01ExpandingCards.png);"
  >
    <div
      class="day"
    >
      Day 1
    </div>
    <div
      class="project__info"
    >
      <h2>
        Expanding Cards
      </h2>
      <button>
        <a
          href="/01ExpandingCards"
        >
          Live Demo
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
